### DISCUSSION TITLE
Docker image in Docker hub

### DISCUSSION THREAD

**Discussion by mmol67** [TracksApp/tracks#2824] [General]
Hello. I've been looking for a GTD selfhosted solution and I've found that Tracks could be for me.

The thing is that I see https://hub.docker.com/r/tracksapp/tracks/, but I don't know how to use it.

Is  MariaDB integrated and db created? How should I use the image? I've read the instructions to install the app in the main site, but it looks complex for me (newbie in docker/selfhosted, and using Portainer stacks). 

It could be a good idea to describe how to use the image, a docker-compose example, a explanation of environment variables, etc in the docker hub page. It would ease the setup for new users.

Regards

**Comment by ZeiP** [comment]
For a simple installation in localhost you can use the Docker Compose instructions at https://github.com/TracksApp/tracks/blob/master/doc/installation.md#docker-installation-using-docker-compose . There's also an instruction set below it for the separate container solution, but that requires creating the DB and user by hand etc.

I agree it'd be a good idea to check the instructions to check if they're as easy as possible!

**Reply by mmol67** [reply]
Novice docker users like me are used to simply having a docker-compose file and that's it. It would be nice to have an image with everything integrated, or that the tracks container would check if the bd exists and if not, create it, at startup.
In any case, thank you very much. I will try to get it up and running tomorrow.

**Reply by agryson** [reply]
I am also hoping to use a compose file, as an example, [Vikunja has this](https://vikunja.io/docs/docker-walkthrough/#create-all-necessary-files) which had me up and running in minutes.
I'll be trying to use that as an example to set up tracks but as a novice this is much more time consuming and error prone.

**Reply by pmbrandvold** [reply]
It appears that the setup script runs "docker-compose" but docker compose is now called with "docker compose" as a native docker install command. I'm getting an error trying to run this with an old docker compose install and the latest version of docker installed on my test machine. Do we need a specific version of compose?
